drivers/timer/stm32_lptim_timer.c

```diff
@@ -314,7 +314,7 @@ uint32_t z_timer_cycle_get_32(void)
 	lp_time += accumulated_lptim_cnt;
 
 	/* convert lptim count in a nb of hw cycles with precision */
-	uint64_t ret = lp_time * (sys_clock_hw_cycles_per_sec() / LPTIM_CLOCK);
+	uint64_t ret = (uint64_t)lp_time * (sys_clock_hw_cycles_per_sec() / LPTIM_CLOCK);
```

Comment:
This doesn't look correct to me, but it was already the case before and not due to to you change. If the computation has to be done with precision as in the comment, it should be:
`uint64_t ret = ((uint64_t)lp_time * sys_clock_hw_cycles_per_sec()) / LPTIM_CLOCK;`

Basically just like the other change. Also note that `sys_clock_hw_cycles_per_sec` is equivalent to `CONFIG_SYS_CLOCK_TICKS_PER_SEC` when `CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME` is not defined, which is the case for this driver.

Reply:
Ok,
I did not find any error when testing (10x) tests/kernel/timer/timer_api and tests/kernel/timer/timer_monotonic on nucleo stm32wb55 (when lptim enabled)